lib/rubocop/cop/style/yoda_condition.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for Yoda conditions where the two parts of an expression
+      # are reversed from the typical order in a conditional statement.
+      #
+      # @example
+      #
+      #   # bad
+      #   99 == foo
+      #   "bar" == foo
+      #
+      # @example
+      #
+      #   # good
+      #   foo == 99
+      #   foo == "bar"
+      class YodaCondition < Cop
+        MSG = 'Reverse the order in conditional statement: `%s` == `%s`.'.freeze
+
+        def on_send(node)
+          return unless comparison_operator?(node)
+
+          register_offense(node) if yoda_condition?(node)
+        end
+
+        private
+
+        def comparison_operator?(node)
+          return false unless node.children.count == 3
+
+          RuboCop::AST::Node::COMPARISON_OPERATORS.include?(node.children[1])
```

Comment:
There's a `SendNode` extension. Use it for less obscure destructuring:

```ruby
node.children[1]
```

vs.

```ruby
node.method_name
```